designdocs/provider_contexts.md

```diff
@@ -0,0 +1,401 @@
+        pub struct ProviderRegistrationPayload<T: Config> {
+            /// Default name for provider or application.
+            pub name: BoundedVec<u8, T::MaxProviderNameSize>,
+
+            /// A list of pre-approved logo image hashes (blake2_256).
+            pub logo_hashes: BoundedVec<[u8; 32], T::MaxApprovedLogoCount>,
+
+            /// Localized translations of the name, keyed by BCP 47 language code.
+            pub translations: BoundedBTreeMap<
+                BoundedVec<u8, T::MaxLanguageCodeSize>, 
+                BoundedVec<u8, T::MaxProviderNameSize>, 
+                T::MaxTranslations,
+            >,
+        }
+
+        #[pallet::call_index(0)]
+        pub fn propose_to_be_provider(
+            origin: OriginFor<T>,
+            payload: ProviderRegistrationPayload,
+        ) -> DispatchResultWithPostInfo {
+            // Implementation details...
+        }
+    ```
+
+2. `propose_to_be_provider` to will insert hashes into the `ApprovedLogoHashes` storage map.
+3. Introduce a new extrinsic `propose_to_add_application` which work in similar way to `propose_to_be_provider` but will be used for adding or updating application contexts.
+
+    ```rust
+        #[pallet::call_index(1)]
+        pub fn propose_to_add_application(
+            origin: OriginFor<T>,
+            payload: ProviderRegistrationPayload,
+        ) -> DispatchResultWithPostInfo {
+            // Implementation details...
+            // This can internally call same logic as `propose_to_be_provider` for consistency
+            // and will also handle the `ProviderToApplicationRegistryEntry` storage map.
+            // It will also handle the `ApprovedLogoHashes` storage map.
+        }
+    ```
+
+    Note:
+    - The same extrinsic should be able to used to proposing new image/logo hashes when an existing application context needs to be updated.
+4. `propose_to_add_application` will insert or update the `ProviderToApplicationRegistryEntry` with the current (numeric) `ApplicationIndex` and `ProviderRegistryEntry`.
+
+    ```rust
+        // Example of how the entry might look like
+        let application_entry = ProviderRegistryEntry {
+            default_name: application_name,
+            localized_names: BTreeMap::new(), // Initially empty, can be updated later
+            default_logo_250_100_png_bytes: BoundedVec::default(), // Initially empty, can be updated later
+            localized_logo_250_100_png_bytes: BTreeMap::new(), // Initially empty, can be updated later
+        };
+
+        ProviderToApplicationRegistryEntry::<T>::insert(provider_id, application_identifier, application_entry);
+    ```
+
+5. `propose_to_add_application` will also insert the logo hashes into the `ApprovedLogoHashes` storage map.
+6. Introduce a new extrinsic to `update_name` for updating an existing application context (post goveranance registration) provided with new name  `ProviderRegistryEntry` with the `ApplicationIndex`.
```

Comment:
For application updates post governance, we introduce 3 extrinsics to update `logo` and `localized logos` , for name and local updates one has to go via `governance` like before